blog/vscode-remote-development-setup-and-usage.mdx

```diff
@@ -0,0 +1,260 @@
+---
+title: "VSCode Remote Development: Setup and usage"
+authors: [gaetan]
+date: 2023-05-02T12:00
+hide_table_of_contents: true
+---
+
+# VSCode Remote Development: Setup and usage
+
+Every developer has their own preferences for their setup, whether it be
+software or hardware. I’ve been a remote developer since 2018, and since then
+I’ve tried various ways to organize my workspace at home, that would fit both my
+work and personal times.
+
+A typical remote developer kit will often include a laptop and one or more
+monitors. In my case since I already had a multi-monitor setup for my personal
+desktop PC, all I needed was to come up with a way to homogenize and make it
+work with the company’s provided laptop.
+
+## My current setup: **VS Code Remote Development**
+
+### What is it ?
+
+[VS Code Remote Development](https://code.visualstudio.com/docs/remote/remote-overview)
+is an extension pack for Microsoft’s widely used IDE VS Code. It allows
+developers to “use a container, remote machine, or the Windows Subsystem for
+Linux (WSL) as a full-featured development environment”. What it means, is that
+you will have your source code located on a single place, and you will be able
+to update it seamlessly from another system using VS Code.
+
+_Note: see our
+[blog post on installing a WSL development environment](https://developers.front-commerce.com/blog/wsl-development-environment)
+if you're interested_
+
+The requirements are:
+
+- For the client, pretty much anything capable of running VS Code _(with special
+  needs for containers for Microsoft Windows due to Docker’s own requirements)_
+- For the host, a wide range of systems are supported. If you’re not using a
+  “mainstream” OS or distribution, you might want to check the requirements
+  section of the
+  [extension pack’s Marketplace page](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.vscode-remote-extensionpack).
+
+In this blog post, we’ll only talk about the SSH connections, so you will need
+to ensure that your host has an SSH Host set up. If you don’t, you can follow
+[Microsoft’s guide on how to setup one](https://code.visualstudio.com/docs/remote/troubleshooting#_installing-a-supported-ssh-server).
+For example, on Ubuntu, it is as simple as running the command
```

Comment:
**nitpick:** I usually try (but it requires attention and rigor!) to ban words such as "simple" or "just have to" from written content to prevent implying a judgment. It can often just be removed without changing the meaning.

See https://developers.google.com/style/word-list#letter-s for other useful things
> simple, simply
> What might be simple for you might not be simple for others. Try eliminating this word from the sentence because usually the same meaning can be conveyed without it.

**suggestion:** 
> For example, on Ubuntu, you can run the command

It is used at several other places in this post, so if you want to go this path it would be useful to do a re-reading dedicated to looking for those.

Reply:
That's a very good remark actually, thank you.

Will do the re-read!